src/CollectdWinService/config/WriteHttpPluginConfig.cs

```diff
@@ -54,6 +54,27 @@ public int MaxIdleTime
                 set { base["MaxIdleTime"] = value; }
             }
 
+            [ConfigurationProperty("UserName", IsRequired = false)]
+            public string UserName
+            {
+                get { return (string)base["UserName"]; }
+                set { base["UserName"] = value; }
+            }
+
+            [ConfigurationProperty("Password", IsRequired = false)]
+            public string Password
+            {
+                get { return (string)base["Password"]; }
+                set { base["Password"] = value; }
+            }
+
+            [ConfigurationProperty("UseSafeNames", IsRequired = false)]

```

Comment:
Instead of UseSafeNames as a boolean, can you have following new configuration?

```
<WindowsPerformanceCounter>
   ...
   <ReplacePluginInstance Enable="true" SafeChars="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789.:-_" ReplaceWith="_" />
   ...
</WindowsPerformanceCounter>